src/configly/utilities.py

```diff
@@ -0,0 +1,32 @@
+def quote_string(value: str):
+    r"""Add syntactically-correct quotation characters to a string value.
+
+    >>> quote_string("foo")
+    '"foo"'
+
+    >>> quote_string("'bar'")
+    "'bar'"
+
+    >>> quote_string("ba'z")
+    '"ba\'z"'
+
+    >>> quote_string('''q"u'x''')
+    '\'q"u\'x\''
+    """
+    if value.startswith('"') and value.endswith('"'):
+        return value
+
+    if value.startswith("'") and value.endswith("'"):
+        return value
+
+    if '"' not in value:
+        return f'"{value}"'
+
+    if "'" not in value:
+        return f"'{value}'"
+
+    # fmt: off
+    value = value.replace("'", "\'")
+    # fmt: on
+
+    return f"'{value}'"
```

Comment:
should this just return value since we add the enclosing `'` on L26?

Reply:
we still want to quote the whole thing and return that - we escape any `'` inside value and then wrap it with our own `'` quotes to finish quoting it.

Reply:
yeah but isnt that what this is doing
```
    if "'" not in value:
        return f"'{value}'"
```

Reply:
this is for a later case - a case where the above is False.

for example, imagine a string like `q"u'x`:

1. we first escape the single quote (L29) - `q"u\'x`
2. then we surround it with single quotes (L32) - `'q"u\'x'`

Reply:
ah i see now. I wasnt reading the `return` statements before